### DISCUSSION TITLE
Add new record if not duplicate

### DISCUSSION THREAD

**Discussion by evgrekov** [pocketbase/pocketbase#5249] [Q&A] [answered]
Hi! I need some advice. One of my most frequent operations is adding a new row to a table, provided it is not a duplicate.

I use a two-step approach: in a try block, I search for the record, and if it is not found (returns an error), I add the record in the catch block.

```
try {
  await pb.collection('shop').getFirstListItem(`customer="${customer}"`);
} catch {
  await pb.collection('shop').create(data);
}

```

Is this approach optimal? Can it be made more efficient in a single query? I would also appreciate other tips on optimization and generally improving code quality. Thank you.

**Comment by ganigeorgiev** [comment, marked as answer]
It is not clear in what context the above code executes and why is needed, bit in general you don't have to manually perform duplicate checks and instead create a unique index constraint for your field from the _"Unique constraints and indexes"_ section in the collection "Fields" tab and then the record create operation will fail on its own with validation error in case of previously inserted record.
